src/nupic/engine/RegionImplFactory.cpp

```diff
@@ -139,8 +139,7 @@ RegionImpl *RegionImplFactory::deserializeRegionImpl(const std::string nodeType,
 
 
 Spec *RegionImplFactory::getSpec(const std::string nodeType) {
-  std::map<std::string, std::shared_ptr<RegisteredRegionImpl>>::iterator it;
-  it = regionTypeMap.find(nodeType);
+  auto it = regionTypeMap.find(nodeType);
```

Comment:
good old auto ;)

Reply:
> ` std::map<std::string, std::shared_ptr<RegisteredRegionImpl>>::iterator it;`
I thought you would like that.  The bug is the `>>`.  There must be a space between them.  `auto` is simpler but it is now difficult to tell what the data type is for resulting item.  In this case it is `RegisteredRegionImpl` but with auto you now have to go to the .hpp, and find the typedef to determine that....unless you are using MSCV where you just put the cursor on `it` and it tells you the type.